Entity/Enemy/g_enemy.cpp

```diff
@@ -19,15 +19,19 @@
 #include "g_enemy_states.h"
 #include "../../input_handler.h"
 
-Enemy::Enemy(const int id, glm::vec3 initialPosition)
-    : MovingEntity(id, ET_ENEMY),
+Enemy::Enemy(const std::vector<Property>& properties)
+    : MovingEntity(ET_ENEMY),
       m_AnimationState(ANIM_STATE_IDLE),
       m_EllipsoidCollider() {
     m_pStateMachine = new StateMachine(this);
     m_pStateMachine->SetCurrentState(EnemyIdle::Instance());
+   
+    // We want position and, if applicable, a target.
+    BaseGameEntity::GetProperty<glm::vec3>(properties, "origin", &m_Position);
+    // FIX: Mem Leak on exit if target is not being set.
+    BaseGameEntity::GetProperty<std::string>(properties, "target", &m_Target);
```

Comment:
this just came to me. target is being overloaded here, since the steeringBehaviour also has a target that the entity moves towards...